args/src/main/java/org/openjdk/skara/args/MultiCommandParser.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.args;
+
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class MultiCommandParser {
+    private final String programName;
+    private final String defaultCommand;
+    private final Map<String, Command> subCommands;
+
+    public MultiCommandParser(String programName, List<Command> commands) {
+        var defaults = commands.stream().filter(Default.class::isInstance).collect(Collectors.toList());
+        if (defaults.size() != 1) {
+            throw new IllegalArgumentException("Expecting exactly one default command");
+        }
+        this.defaultCommand = defaults.get(0).name();
+
+        this.programName = programName;
+        this.subCommands = commands.stream()
+                                   .collect(Collectors.toMap(
+                                           Command::name,
+                                           Function.identity()));
+        this.subCommands.put("help", helpCommand());
+    }
+
+    private Command helpCommand() {
+        return new Command("help", "print a help message", args -> showUsage());
+    }
+
+    public Executable parse(String[] args) {
+        if (args.length != 0) {
```

Comment:
Super tiny comment, and you can fix this before integrating, but I always prefer a positive check. In this case it would be `args.length > 0` to show that we can read the first argument.